components/dps310/src/dps310.c

```diff
@@ -0,0 +1,455 @@
+{
+    esp_err_t err = ESP_FAIL;
+
+    if (dev == NULL)
+    {
+        err = ESP_ERR_INVALID_ARG;
+        goto fail;
+    }
+    if (addr != DPS310_I2C_ADDRESS_0 && addr != DPS310_I2C_ADDRESS_1)
+    {
+        ESP_LOGE(TAG, "Invalid I2C address: expected %0X or %0X, got %0X",
+                 DPS310_I2C_ADDRESS_0, DPS310_I2C_ADDRESS_1, addr);
+        err = ESP_ERR_INVALID_ARG;
+        goto fail;
+    }
+
+    dev->i2c_dev.port = port;
+    dev->i2c_dev.addr = addr;
+    dev->i2c_dev.cfg.sda_io_num = sda_gpio;
+    dev->i2c_dev.cfg.scl_io_num = scl_gpio;
+#if HELPER_TARGET_IS_ESP32
+    dev->i2c_dev.cfg.master.clk_speed = DPS310_I2C_FREQ_MAX_ESP_IDF_HZ;
+#endif
+
+    ESP_LOGD(TAG, "Port: %u SCL: GPIO%i SDA: GPIO%i",
+             port,
+             dev->i2c_dev.cfg.scl_io_num,
+             dev->i2c_dev.cfg.sda_io_num);
+    err = i2c_dev_create_mutex(&dev->i2c_dev);
+    if (err != ESP_OK)
+    {
+        ESP_LOGE(TAG, "i2c_dev_create_mutex(): %s", esp_err_to_name(err));
+        goto fail;
+    }
+
+fail:
+    return err;
+
+}
+
+esp_err_t dps310_free_desc(dps310_t *dev)
+{
+    esp_err_t err = ESP_FAIL;
+
+    err = i2c_dev_delete_mutex(&dev->i2c_dev);
+    if (err != ESP_OK)
+    {
+        ESP_LOGW(TAG, "i2c_dev_delete_mutex(): %s", esp_err_to_name(err));
+        goto fail;
+    }
+fail:
+    return err;
+}
+
+esp_err_t dps310_init(dps310_t *dev, dps310_config_t *config)
+{
+    uint8_t reg_value = 0;
+    esp_err_t err = ESP_FAIL;
+
+    if (dev == NULL)
+    {
+        ESP_LOGE(TAG, "dps310_init(): invalid dev");
+        err = ESP_ERR_INVALID_ARG;
+        goto fail;
+    }
+    if (config == NULL)
+    {
+        ESP_LOGE(TAG, "dps310_init(): invalid config");
+        err = ESP_ERR_INVALID_ARG;
+        goto fail;
+    }
+
+    err = _read_reg(&dev->i2c_dev, DPS310_REG_ID, &reg_value);
+    if (err != ESP_OK)
+    {
+        ESP_LOGE(TAG, "Device not found");
+        goto fail;
+    }
+
+    dev->prod_id = (DPS310_REG_ID_PROD_MASK & reg_value) >> 4;
+    dev->prod_rev = DPS310_REG_ID_REV_MASK & reg_value;
+    ESP_LOGD(TAG, "prod_id: 0x%x prod_rev: 0x%x", dev->prod_id, dev->prod_rev);
+    if (dev->prod_id != DPS310_PROD_ID)
+    {
+        ESP_LOGE(TAG, "Invalid prod ID: expected: 0x%x (DPS310) got: 0x%x", DPS310_PROD_ID, dev->prod_id);
+        err = ESP_FAIL;
+        goto fail;
+    }
+
+    err = dps310_reset(dev);
+    if (err != ESP_OK)
+    {
+        ESP_LOGE(TAG, "dps310_reset(): %s", esp_err_to_name(err));
+        goto fail;
+    }
+
+    err = dps310_quirk(dev);
+    if (err != ESP_OK)
+    {
+        ESP_LOGE(TAG, "dps310_quirk(): %s", esp_err_to_name(err));
+        goto fail;
+    }
+
+    err = ESP_OK;
+
+fail:
+    return err;
+}
+
+esp_err_t dps310_get_pm_rate(dps310_t *dev, uint8_t *value)
+{
+    uint8_t reg_value = 0;
+    esp_err_t err = ESP_FAIL;
+
+    CHECK_ARG(dev && value);
+
+    I2C_DEV_TAKE_MUTEX(&dev->i2c_dev);
```

Comment:
maybe `_read_reg_mask()`

Reply:
should be. this one is the first function  i wrote for the driver and later i wrote `_read_reg_mask`.